reader.go

```diff
@@ -64,11 +69,59 @@ type ReaderConfig struct {
 	// of messages from kafka.
 	MaxWait time.Duration
 
+	// ReadLagInterval sets the frequency at which the reader lag is updated.
+	// Setting this field to a negative value disables lag reporting.
+	ReadLagInterval time.Duration
+
 	// If not nil, specifies a logger used to report internal changes within the
 	// reader.
 	Logger *log.Logger
 }
 
+// ReaderStats is a data structure returned by a call to Reader.Stats that exposes
+// details about the behavior of the reader.
+type ReaderStats struct {
+	Dials      int64 `metric:"kafka.reader.dial.count"      type:"counter"`
+	Fetches    int64 `metric:"kafak.reader.fetch.count"     type:"counter"`
+	Messages   int64 `metric:"kafka.reader.message.count"   type:"counter"`
+	Bytes      int64 `metric:"kafka.reader.message.bytes"   type:"counter"`
+	Rebalances int64 `metric:"kafka.reader.rebalance.count" type:"counter"`
+	Timeouts   int64 `metric:"kafka.reader.timeout.count"   type:"counter"`
+	Errors     int64 `metric:"kafka.reader.error.count"     type:"counter"`
+
+	DialTime DurationStats `metric:"kafka.reader.dial.seconds"`
+	ReadTime DurationStats `metric:"kafka.reader.read.seconds"`
+	WaitTime DurationStats `metric:"kafka.reader.wait.seconds"`
+
+	Offset        int64 `metric:"kafka.reader.offset"          type:"gauge"`
```

Comment:
Should offset be a gauge or counter?

Reply:
Gauge seems better because the reader exposes method to change the offset that it's reading from, so it may go backward sometimes.

Reply:
I see, makes sense.  I wish Datadog were better at graphing gauge types then :-(.